contracts/src/NFTMinter.sol

```diff
@@ -2,59 +2,190 @@
 pragma solidity ^0.8.20;
 
 import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Strings.sol";
 
-    
-contract NFTMinter is ERC1155 {
+import "src/libraries/StringBytes32.sol";
+
+/**
+ * @title NFTMinter
+ * @dev A contract for minting NFTs that represent domain ownership
+ * Works with a multi-chain domain registration system
+ */
+contract NFTMinter is ERC1155, Ownable {
     using Strings for uint256;
-    
-    
-    // Mapping to store token names
-    mapping(uint256 => string) private _tokenNames;
+    using StringBytes32 for string;
+    using StringBytes32 for bytes32;
     
     // Counter for token IDs
-    uint256 private _tokenIdCounter;
+    uint256 public s_tokenCounter;
     
-    // Event emitted when a new NFT is minted
-    event NFTMinted(uint256 indexed tokenId, address indexed to, string name);
+    // Mappings for domain tracking
+    mapping(bytes32 => uint256) public s_domainNameToTokenId;
+    mapping(uint256 => bytes32) public s_tokenIdToDomainName;
+    mapping(bytes32 => bool) public s_isDomainMintable;
+    mapping(bytes32 => address) public s_domainNameToOwner;
+    mapping(bytes32 => bool) public s_isDomainNFTMinted;
     
-    constructor() ERC1155("ipfs://") {
+    // Events
+    event NFTMinted(uint256 indexed tokenId, address indexed to, bytes32 domainNameHash, string domainName);
+    event DomainOwnerSet(bytes32 indexed domainNameHash, address owner, string domainName);
+    event DomainMintableStatusSet(bytes32 indexed domainNameHash, bool isMintable, string domainName);
+    event DomainNFTMintedStatusSet(bytes32 indexed domainNameHash, bool isMinted, string domainName);
+    
+    /**
+     * @dev Constructor sets the IPFS base URI and initializes ownership
+     */
+    constructor() ERC1155("ipfs://") Ownable(msg.sender) {
```

Comment:
can we make this url "ipfs xxx" configurable? or support more? 

most likely we want to support other providers like flow native, arweave, irys etc